framework/auth/campaigns.py

```diff
@@ -0,0 +1,47 @@
+import httplib as http
+
+from framework.exceptions import HTTPError
+
+from website import mails
+from website.util import web_url_for
+
+VALID_CAMPAIGNS = (
+    'prereg',
+)
+
+EMAIL_TEMPLATE_MAP = {
+    'prereg': mails.CONFIRM_EMAIL_PREREG
+}
+
+def email_template_for_campaign(campaign, default=None):
+    if campaign in VALID_CAMPAIGNS:
+        try:
+            return EMAIL_TEMPLATE_MAP[campaign]
```

Comment:
Aside from other refactoring options mentioned, this could be shortened to
```
template = EMAIL_TEMPLATE_MAP.get(campaign, default)
if template is None:
    raise DeveloperError
return template
```

(Unless there's a reason why VALID_CAMPAIGNS and EMAIL_TEMPLATE_MAP would be decoupled; I just assume that there wouldn't be a template defined if the campaign wasn't valid)